openedx/core/djangoapps/programs/tests/test_utils.py

```diff
@@ -1211,8 +1275,11 @@ def test_fetching_program_discounted_price_as_anonymous_user(self):
             body=json.dumps(mock_discount_data),
             content_type='application/json'
         )
+        user = AnonymousUserFactory()
+        user.courseenrollment_set = CourseEnrollment.objects.none()
```

Comment:
Is there a empty CourseEnrollment factory or something similar? What happens if you set this to `None`?

Reply:
`courseenrollment_set` and `courseentitlement_set` need to be empty query sets as opposed to just `None` because the one-click eligibility logic performs filters on them, which cannot be performed on NoneType objects.
I can look into a more neutral empty query set creation method though.

Reply:
The django model `EmptyQuerySet` can't be instantiated, so I think this is the best way to set these two user properties.

Reply:
Why do you need to set these at all? Shouldn't they always return nothing anyway since there shouldn't be any entitlements in the database for the anonymous user?

Reply:
When I don't explicitly set them I get this error when running the test
`AttributeError: 'AnonymousUser' object has no attribute 'courseenrollment_set'`

Reply:
Is there another factory we could use to get a user object that is more similar to one that would actually be passed through this codepath? Only authenticated users should be hitting this code path.

Reply:
Have you looked at `UserFactory` in `from student.tests.factories import UserFactory`